packages/backend/src/controllers/schedulerController.ts

```diff
@@ -0,0 +1,124 @@
+import {
+    ApiErrorPayload,
+    ApiScheduledJobsResponse,
+    ApiSchedulerAndTargetsResponse,
+    UpdateSchedulerAndTargetsWithoutId,
+} from '@lightdash/common';
+import { Delete } from '@tsoa/runtime';
+import express from 'express';
+import {
+    Body,
+    Controller,
+    Get,
+    Middlewares,
+    OperationId,
+    Patch,
+    Path,
+    Request,
+    Response,
+    Route,
+    SuccessResponse,
+} from 'tsoa';
+import { SchedulerService } from '../services/SchedulerService/SchedulerService';
+import { schedulerService } from '../services/services';
+import { allowApiKeyAuthentication, isAuthenticated } from './authentication';
+
+@Route('/api/v1/schedulers')
+@Response<ApiErrorPayload>('default', 'Error')
+export class SchedulerController extends Controller {
+    /**
+     * Get a scheduler
+     * @param schedulerUuid The uuid of the scheduler to update
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('{schedulerUuid}')
+    @OperationId('getScheduler')
+    async get(
+        @Path() schedulerUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiSchedulerAndTargetsResponse> {
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: await schedulerService.getScheduler(
+                req.user!,
+                schedulerUuid,
+            ),
+        };
+    }
+
+    /**
+     * Update a scheduler
+     * @param schedulerUuid The uuid of the scheduler to update
+     * @param req express request
+     * @param body the new scheduler data
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('201', 'Updated')
+    @Patch('{schedulerUuid}')
+    @OperationId('updateScheduler')
+    async patch(
+        @Path() schedulerUuid: string,
+        @Request() req: express.Request,
+        @Body() body: any, // TODO: It should be UpdateSchedulerAndTargetsWithoutId but tsoa returns an error
```

Comment:
what's the error here?

Reply:
The library has an internal error as it can't interpret our type. There are a few open tickets for similar issues in the library github.